editor/src/document/document_file.rs

```diff
@@ -842,10 +847,12 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
 							}
 							DocumentResponse::LayerChanged { path } => responses.push_back(LayerChanged(path.clone()).into()),
 							DocumentResponse::CreatedLayer { path } => {
-								self.layer_metadata.insert(path.clone(), LayerMetadata::new(false));
+								if !self.layer_metadata.contains_key(path) {
```

Comment:
I think this should be an error right?
could we maybe add a log::warn! in the else case? I don't think this should ever not be true

Reply:
This should never be true, I just misread the PasteIntoFolder as populating the layermetadata before the InsertData (when actually it is push_front)